drivers/serial/uart_imx.c

```diff
@@ -0,0 +1,544 @@
+/*
+ * Copyright (c) 2018 Diego Sueiro <diego.sueiro@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+/**
+ * @brief Driver for UART on NXP IMX family processor.
+ *
+ * For full serial function, use the USART controller.
+ *
+ */
+
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <misc/__assert.h>
+#include <board.h>
+#include <init.h>
+#include <uart.h>
+#include <uart_imx.h>
+
+#define DEV_CFG(dev) \
+	((const struct uart_device_config * const)(dev)->config->config_info)
+#define DEV_DATA(dev) \
+	((struct uart_imx_dev_data_t * const)(dev)->driver_data)
+#define UART_STRUCT(dev) \
+	((UART_Type *)(DEV_CFG(dev))->base)
+
+/* Device data structure */
+struct uart_imx_dev_data_t {
```

Comment:
This is really a config structure, not a data structure. Can you please rename it to `struct uart_imx_dev_config`